### PR TITLE
feat: Hosea/ext 192 add twenty crm support

### PR DESCRIPTION
**Pull request by mungaihosea**
## Describe your changes
Added support for Twenty CRM

## Issue ticket number and link
EXT-192
https://linear.app/nango/issue/EXT-192/add-twenty-crm-support

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/shared/providers.yaml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# yaml-language-server: $schema=./../../scripts/validation/providers/schema.json
```
Comment (khaliqgant) [inline packages/shared/providers.yaml]: Why was this removed?

Comment (mungaihosea) [reply packages/shared/providers.yaml]: I have reinstated the comment

File: packages/shared/providers.yaml [hunk 2]
```diff
@@ -5578,6 +5578,27 @@ twinfield:
...
+        headers:
+            Authorization: Bearer ${apiKey}
+        retry:
+            after: 'Retry-After'
+        verification:
+            method: GET
+            endpoint: /companies
+    docs: https://docs.nango.dev/integrations/all/twenty-crm
+    credentials:
+        apiKey:
```
Comment (hassan254-prog) [inline packages/shared/providers.yaml:5656]: might be good to include a pattern and example in here.

File: docs-v2/integrations/all/twenty-crm.mdx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+| [Proxy requests](/integrate/guides/proxy-requests-to-an-api) | ✅ |
+| [Receive webhooks](/integrate/guides/receive-webhooks-from-an-api) | 🚫 (time to contribute: &lt;48h) |
+
+<Tip>We can implement missing features in &lt;48h, just ask for it in the [community](https://nango.dev/slack).</Tip>
+
+## Getting started
+-   [API docs](https://app.twenty.com/settings/developers)
+-   [Generate API keys](https://app.twenty.com/settings/developers)
+-   [Authentication](https://twenty.com/developers/rest-api/core#/)
+-   [API docs](https://app.twenty.com/settings/developers)
```
Comment (hassan254-prog) [inline docs-v2/integrations/all/twenty-crm.mdx]: It looks like you may have included a duplicate link to the docs, I'm not sure if that was intentional. Additionally, you could consider adding a link to the documentation on rate limiting.

File: packages/shared/providers.yaml (hunk 2 above)
Comment (mungaihosea) [reply packages/shared/providers.yaml:5656]: couldn't find any information on rate limiting of Twenty CRM APIs

Comment (mungaihosea) [reply packages/shared/providers.yaml:5597]: I have removed the duplicate link

File: docs-v2/integrations/all/twenty-crm.mdx (hunk 3 above)
Comment (mungaihosea) [reply docs-v2/integrations/all/twenty-crm.mdx]: the duplicate link has been removed

File: packages/shared/providers.yaml (hunk 2 above)
Comment (hassan254-prog) [reply packages/shared/providers.yaml:5656]: Some providers may not include this, so you can omit it if not present in the docs. Please check [comment](https://github.com/NangoHQ/nango/pull/2946#discussion_r1830428059)

File: packages/shared/providers.yaml [hunk 4]
```diff
@@ -5578,6 +5578,27 @@ twinfield:
     disable_pkce: true
     docs: https://docs.nango.dev/integrations/all/twinfield
 
+twenty-crm:
+    display_name: Twenty CRM
+    categories:
+        - crm
+    auth_mode: API_KEY
+    proxy:
+        base_url: https://api.twenty.com/rest
```
Comment (bodinsamuel) [inline packages/shared/providers.yaml:5646]: @khaliqgant @hassan254-prog  wondering about open-source project like this. They can be hosted or cloud, should we set the base_url to the cloud or have a connectionConfig or create two provider?

File: packages/shared/providers.yaml [hunk 5]
```diff
@@ -5578,6 +5578,27 @@ twinfield:
...
+            after: 'Retry-After'
+        verification:
+            method: GET
+            endpoint: /companies
+    docs: https://docs.nango.dev/integrations/all/twenty-crm
+    credentials:
+        apiKey:
+            type: string
+            title: API Key
+            description: The API key for your Twenty CRM account
```
Comment (bodinsamuel) [inline packages/shared/providers.yaml:5659]: If you can add pattern for verification 👍🏻

**Review by bodinsamuel — APPROVED** [review]

File: packages/shared/providers.yaml (hunk 4 above)
Comment (khaliqgant) [reply packages/shared/providers.yaml:5646]: Good point. I think we should have a connectionConfig param actually for this.

```
base_url: https://${connectionConfig.domain}/rest
```

File: packages/shared/providers.yaml (hunk 5 above)
Comment (khaliqgant) [reply packages/shared/providers.yaml:5659]: Added with https://github.com/NangoHQ/nango/pull/2946/commits/151def42a6f0c0736e153898436c09f0f203ab2f

File: packages/shared/providers.yaml (hunk 4 above)
Comment (khaliqgant) [reply packages/shared/providers.yaml:5646]: Done with https://github.com/NangoHQ/nango/pull/2946/commits/151def42a6f0c0736e153898436c09f0f203ab2f

**Review by khaliqgant — APPROVED** [review]
